src/query/expression/src/evaluator.rs

```diff
@@ -973,6 +1025,77 @@ impl<'a, Index: ColumnIndex> ConstantFolder<'a, Index> {
                     new_domain,
                 )
             }
+            Expr::FunctionCall {
+                span,
+                id,
+                function,
+                generics,
+                args,
+                return_type,
+            } if function.signature.name == "and_filters" => {
+                let (mut args_expr, mut args_domain) = (Vec::new(), Some(Vec::new()));
+                for arg in args {
+                    let (expr, domain) = self.fold_once(arg);
+                    args_expr.push(expr);
+                    args_domain = args_domain.zip(domain).map(|(mut domains, domain)| {
+                        domains.push(domain);
+                        domains
+                    });
+                }
```

Comment:
we need to fold the arguments along in the following loop, so as to shortcut the domain calculation when a `!has_true && has_false` is met. Otherwise, like the current implementation, `and_filters(false, x/0)` will never get folded because `args_domain` will be equal `None`.